elasticdl/pkg/ps/parameter_test.go

```diff
@@ -27,3 +28,39 @@ func TestParameterInit(t *testing.T) {
 	assert.Equal(t, p.GetNonEmbeddingParam("t2").Dim, d2)
 	assert.Nil(t, p.GetNonEmbeddingParam("t3"))
 }
+
+func TestParameterDeserialize(t *testing.T) {
+	var modelPB proto.Model
+	modelPB.Version = int32(1)
+
+	i1 := []int64{1, 3, 5}
+	d1 := []int64{3, 2}
+	v1 := []float32{1.0, 2.0, 3.0, 4.0, 5.0, 6.0}
+	t1 := common.Tensor{"e1", v1, d1, i1}
+	p1 := common.SerializeTensor(&t1)
+	modelPB.Param = append(modelPB.Param, p1)
+
+	var epb proto.EmbeddingTableInfo
+	epb.Name = "e1"
+	epb.Dim = 2
+	epb.Initializer = "zero"
+	modelPB.EmbeddingTableInfo = append(modelPB.EmbeddingTableInfo, &epb)
+
+	p, err := DeserializeModelPB(&modelPB)
+
+	assert.Nil(t, err)
+	assert.Contains(t, p.EmbeddingParam, "e1")
+
+	e1 := p.GetEmbeddingParam("e1")
+	assert.Equal(t, int64(2), e1.Dim)
+	assert.Equal(t, 3, len(e1.EmbeddingVector))
+
+	ev1 := e1.GetEmbeddingVector(1)
```

Comment:
Test these in a loop and reduce the amount of duplicated code?

Reply:
A loop will not reduce the lines of test code here.
It's just a unit test, I think some duplicated code is acceptable.